src/ast.cc

```diff
@@ -403,13 +380,15 @@ void destroy_table(Table* table) {
 static void destroy_module_field(ModuleField* field) {
   switch (field->type) {
     case ModuleFieldType::Func:
-      destroy_func(&field->func);
+      destroy_func(field->func);
+      delete field->func;
```

Comment:
Is "*field->func" a class object? If so, the destructor can call "destroy_func" and this can be simplified to just: "delete field->func".